app/src/candid/candid_parser.c

```diff
@@ -71,54 +75,67 @@ parser_error_t readCandidListNeurons(parser_tx_t *tx, const uint8_t *input, uint
     CHECK_PARSER_ERR(getCandidTypeFromTable(&txn, tx->candid_rootType))
 
     CHECK_PARSER_ERR(readCandidRecordLength(&txn))
-    if (txn.txn_length != 3) {
+    // at least we need to have the 2 non opt fields already defined in the did file
```

Comment:
what if they later decide to remove one of them from the blob?
just asking not sure if this case is defined in #231

Reply:
This check if for the two non optionals that are in the record, this should not be removed, if they are, we need to change the parser. This case is not defined by ICP has one of the cases we should support since is non-optional related.